src/Result.php

```diff
@@ -16,13 +18,22 @@ public function isValid(): bool
         return $this->errors === [];
     }
 
-    public function addError(string $message): void
+    /**
+     * @param string $message
+     * @param int|string|null $key
+     * @return void
+     */
+    public function addError(string $message, $key = null): void
     {
-        $this->errors[] = $message;
+        if ($key !== null) {
+            $this->errors[$key] = $message;
+        } else {
+            $this->errors[] = $message;
+        }
     }
 
     /**
-     * @psalm-return list<string>
+     * @psalm-var list<int|string, string>
```

Comment:
```suggestion
     * @psalm-param list<int|string, string>
```

Reply:
It's actually neither `var`, nor `param`. It's `return`. Thank you for noticing. Fixed. Probably was copy pasted from above.

Reply:
:heavy_check_mark: